crates/hotshot/src/traits/networking/web_server_network.rs

```diff
@@ -929,6 +952,70 @@ impl<TYPES: NodeType + 'static, NetworkVersion: StaticVersionType + 'static>
                     });
                 }
             }
+            ConsensusIntentEvent::PollForUpgradeProposal(view_number) => {
+                // Only start this task if we haven't already started it.
+                let mut cancel_handle = self.inner.upgrade_proposal_task.write().await;
+                if cancel_handle.is_none() {
+                    error!("Starting poll for upgrade proposals!");
+                    let inner = self.inner.clone();
+
+                    // Create sender and receiver for cancelling the task
+                    let (sender, receiver) = unbounded();
+                    *cancel_handle = Some(sender);
+
+                    // Create the new task
+                    async_spawn(async move {
+                        if let Err(e) = inner
+                            .poll_web_server(
+                                receiver,
+                                MessagePurpose::UpgradeProposal,
+                                view_number,
+                                Duration::from_millis(500),
```

Comment:
Yeah, I think this was just copied from the other branches. I didn't spend much time cleaning this up because I think we're planning on completely deleting the `web_server` anyway: #2825

Reply:
Sounds good!